src/Beberlei/Metrics/Collector/CollectorInterface.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Beberlei Metrics
+ *
+ * LICENSE
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this package in the file LICENSE.txt.
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to kontakt@beberlei.de so I can send you a copy immediately.
+ */
+
+namespace Beberlei\Metrics\Collector;
+
+interface CollectorInterface
+{
+    /**
+     * Update one or more counters by some arbitrary amount
+     *
+     * @param string $variable
+     * @param int $value The amount to increment the counter by
+     * @return CollectorInterface
+     */
+    public function measure($variable, $value);
+
+    /**
+     * Increment a counter
+     *
+     * @param string $variable
+     * @return CollectorInterface
+     */
+    public function increment($variable);
+
+    /**
+     * Decrement a counter
+     *
+     * @param string $variable
+     * @return CollectorInterface
+     */
+    public function decrement($variable);
+
+    /**
+     * Record a timing
+     *
+     * @param string $variable
+     * @param int $time The duration of the timing in milliseconds
+     * @return CollectorInterface
+     */
+    public function timing($variable, $time);
+
+    /**
+     * Update a gauge by an arbitrary amount
+     *
+     * @param string $variable
+     * @param int $value
+     * @return CollectorInterface
+     */
+    public function gauge($variable, $value);
```

Comment:
Adding a new method in the interface is a BC break, meaning it would force to bump the major version.